library/src/main/java/jp/co/cyberagent/android/gpuimage/GLTextureView.java

```diff
@@ -0,0 +1,1809 @@
+                }
+                requestPaused = false;
+                requestRender = true;
+                renderComplete = false;
+                glThreadManager.notifyAll();
+                while ((!exited) && paused && (!renderComplete)) {
+                    if (LOG_PAUSE_RESUME) {
+                        Log.i("Main thread", "onResume waiting for !paused.");
+                    }
+                    try {
+                        glThreadManager.wait();
+                    } catch (InterruptedException ex) {
+                        Thread.currentThread().interrupt();
+                    }
+                }
+            }
+        }
+
+        public void onWindowResize(int w, int h) {
+            synchronized (glThreadManager) {
+                width = w;
+                height = h;
+                sizeChanged = true;
+                requestRender = true;
+                renderComplete = false;
+                glThreadManager.notifyAll();
+
+                // Wait for thread to react to resize and render a frame
+                while (!exited && !paused && !renderComplete && ableToDraw()) {
+                    if (LOG_SURFACE) {
+                        Log.i("Main thread", "onWindowResize waiting for render complete from tid=" + getId());
+                    }
+                    try {
+                        glThreadManager.wait();
+                    } catch (InterruptedException ex) {
+                        Thread.currentThread().interrupt();
+                    }
+                }
+            }
+        }
+
+        public void requestExitAndWait() {
+            // don't call this from GLThread thread or it is a guaranteed
+            // deadlock!
+            synchronized (glThreadManager) {
+                shouldExit = true;
+                glThreadManager.notifyAll();
+                while (!exited) {
+                    try {
+                        glThreadManager.wait();
+                    } catch (InterruptedException ex) {
+                        Thread.currentThread().interrupt();
+                    }
+                }
+            }
+        }
+
+        public void requestReleaseEglContextLocked() {
+            shouldReleaseEglContext = true;
+            glThreadManager.notifyAll();
+        }
+
+        /**
+         * Queue an "event" to be run on the GL rendering thread.
+         *
+         * @param r the runnable to be run on the GL rendering thread.
+         */
+        public void queueEvent(Runnable r) {
+            if (r == null) {
+                throw new IllegalArgumentException("r must not be null");
+            }
+            synchronized (glThreadManager) {
+                eventQueue.add(r);
+                glThreadManager.notifyAll();
+            }
+        }
+
+        // Once the thread is started, all accesses to the following member
+        // variables are protected by the glThreadManager monitor
+        private boolean shouldExit;
```

Comment:
Please declare fields before methods.

Reply:
This GLTextureView is based on GLSurfaceView, so I won't change that.
https://android.googlesource.com/platform/frameworks/base/+/master/opengl/java/android/opengl/GLSurfaceView.java

Reply:
I understand. 👍